src/Microsoft.OData.Edm/EdmUtil.cs

```diff
@@ -492,6 +500,82 @@ internal static string FullyQualifiedName(IEdmVocabularyAnnotatable element)
             return null;
         }
 
+        internal static string FullyQualifiedName(IEdmAnnotationsTarget annotationsTarget)

```

Comment:
If the `IEdmAnnotationsTarget` parameter will always contain at least 3 segments, could you at least add some debug asserts to verify the pre-conditions. Otherwise an `IndexOutOfRangeException` exception would be throw if the parameter is null or doesn't contain at least 3 segments for any reason.